From 6b7cb1fcf679d83b22c58dfd5f47cbb12da987b3 Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Tue, 2 Dec 2025 15:16:13 +0800
Subject: [PATCH 42/63] loonggpu: drop LG_ZONE_MANAGED_PAGES

With commit 959b0886256b ("mm: constify zone related test/getter
function"), the argument to `zone_managed_pages()' is now `const'.
The commit was included as part of v6.18.

This causes conftest.sh to fail to detect whether the kernel supports
zone-managed pages, as it would no longer match the function prototype.

Since this conftest was really used to deal with pre-5.1 kernels, it is
most definitely irrelevant for us. Nullify it to fix build with 6.18.

Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 lgkcd/kcd_crat.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/lgkcd/kcd_crat.c b/lgkcd/kcd_crat.c
index 46fb016..b2aab0f 100644
--- a/lgkcd/kcd_crat.c
+++ b/lgkcd/kcd_crat.c
@@ -583,11 +583,7 @@ static int kcd_fill_mem_info_for_cpu(int numa_node_id, int *avail_size,
 	 */
 	pgdat = NODE_DATA(numa_node_id);
 	for (zone_type = 0; zone_type < MAX_NR_ZONES; zone_type++)
-#if defined(LG_ZONE_MANAGED_PAGES)
 		mem_in_bytes += zone_managed_pages(&pgdat->node_zones[zone_type]);
-#else
-		mem_in_bytes += pgdat->node_zones[zone_type].managed_pages;
-#endif
 	mem_in_bytes <<= PAGE_SHIFT;
 
 	sub_type_hdr->length_low = lower_32_bits(mem_in_bytes);
-- 
2.52.0

